modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/side_navigation.es.js

```diff
@@ -651,11 +920,23 @@ SideNavigation.prototype = {
 
 			instance._emit('closedStart.lexicon.sidenav');
 
+			dispatchCustomEvent(
+				document,
+				'closedStart.lexicon.sidenav',
+				instance
+			);
```

Comment:
A pretty obvious smell here is that _every_ time you are emit an event on the instance itself, you _also_ emit an identical duplicate event as a custom event via the `document`. At the very least, that suggests that you should move the `dispatchCustomEvent()` into `_emit()` so that you have a centralized "choke point" where the double emission occurs and you don't have to repeat it in so many places. The other hint that something is amiss is that you have made a three-parameter `dispatchCustomEvent()` function which _always_ takes the same values (`document` and `instance`) for parameters 1 and 3; that's another indication that you could/should just inline it into `_emit()`.

But most of all, I'm interested in hearing why we need to do this double emission at all. Unless I am overlooking it, I can't see anybody in this PR that is subscribed to the custom events on `document`. What's the rationale for doing this?

Reply:
I was looking into using `instance._emit()` at first, but realized the event gets fired on the menu itself. It made it difficult to calculate the padding needed to size `sidenav-content` when two menus are independent of each other (product menu and info panel). 

I'll make the other changes and push them up.

Reply:
Ok, so if I understand you correctly, when you have two (or hypothetically, more) "sidenavs" on the screen at the same time, you want _all_ menus to recalculate in response to an event on any other, and that's why you do the duplicate global broadcast. I am not a huge fan of the redundancy there, in the sense that we should be able to get rid of the "local" emitter and update all existing listeners to use the global one instead. Having both is kind of like having two doorbells on your apartment, one that rings inside your apartment, and the other that anyone in the building can listen to.

Reply:
Yes exactly that. I agree with no redundancy. I'll look into removing one or the other.